photon-ml/src/main/scala/com/linkedin/photon/ml/cli/game/GAMEDriver.scala

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.cli.game
+
+import com.linkedin.photon.ml.util._
+import com.linkedin.photon.ml.avro.data.NameAndTermFeatureSetContainer
+
+import org.apache.spark.SparkContext
+
+/**
+  * Contains common functions for GAME training and scoring drivers.
+  */
+class GAMEDriver(
+    params: FeatureParams with PalDBIndexMapParams,
+    sparkContext: SparkContext,
+    logger: PhotonLogger) {
+
+  import params._
+
+  protected val hadoopConfiguration = sparkContext.hadoopConfiguration
+
+  /**
+    * Builds feature key to index map loaders according to configuration
+    *
+    * @return A map of shard id to feature map loader
+    * @deprecated This function will be removed in the next major version.
+    */
+  protected[game] def prepareFeatureMapsDefault(): Map[String, IndexMapLoader] = {
+    val allFeatureSectionKeys = featureShardIdToFeatureSectionKeysMap.values.reduce(_ ++ _)
+    val nameAndTermFeatureSetContainer = NameAndTermFeatureSetContainer.readNameAndTermFeatureSetContainerFromTextFiles(
+      featureNameAndTermSetInputPath, allFeatureSectionKeys, hadoopConfiguration)
+
+    val featureShardIdToFeatureMapLoader =
+      featureShardIdToFeatureSectionKeysMap.map { case (shardId, featureSectionKeys) =>
+        val featureMap = nameAndTermFeatureSetContainer
+          .getFeatureNameAndTermToIndexMap(featureSectionKeys, featureShardIdToInterceptMap.getOrElse(shardId, true))
+          .map { case (k, v) => Utils.getFeatureKey(k.name, k.term) -> v }
```

Comment:
It looks like that ```featureShardIdToInterceptMap``` now is only needed to prepare default feature maps.

Now we have some API level inconsistency in adding intercept to different feature shard between default index map and PalDB based index map.